### DISCUSSION TITLE
feat: ability to use drizzle-zod validation schemas in expo

### DISCUSSION THREAD

**Discussion by trevorpfiz** [t3-oss/create-t3-turbo#966] [Ideas]
### Describe the feature you'd like to request

I am reading this on the create-t3-turbo README: 
```
Does this pattern leak backend code to my client applications?
No, it does not. The api package should only be a production dependency in the Next.js application where it's served. The Expo app, and all other apps you may add in the future, should only add the api package as a dev dependency. This lets you have full typesafety in your client applications, while keeping your backend code safe.

If you need to share runtime code between the client and server, such as input validation schemas, you can create a separate shared package for this and import it on both sides.
```

Currently, I am referencing the OpenStatus repo where they create the validation schemas in the db package https://github.com/openstatusHQ/openstatus/blob/main/packages/db/src/schema/users/validation.ts. How would I use these in expo if I cannot have `@acme/db` as a dependency?

Do I have to manually create the validators in the validators package and only use `drizzle-zod` in Next.js?

### Describe the solution you'd like to see

Ideally there is a way to use the drizzle-zod validation schemas in expo safely, but I am not sure how if I understand correctly.

**Comment by dBianchii** [comment]
I believe you can do this now by just using @acme/validators?

**Comment by trevorpfiz** [comment]
> I believe you can do this now by just using @acme/validators?

wouldn't we have to install `@acme/db` as a dependency? I'm guessing that is still a problem?

`@acme/validators/index.ts`
```js
import { createInsertSchema, createSelectSchema } from "drizzle-zod";
import type { z } from "zod";

import { user } from "@acme/db/src/schema";

export const insertUserSchema = createInsertSchema(user);

export const selectUserSchema = createSelectSchema(user);

export type InsertUser = z.infer<typeof insertUserSchema>;
export type User = z.infer<typeof selectUserSchema>;
```

**Comment by CarlosDavidAraujo** [comment]
Any solution? I tried to import `@acme/db` into validators and this causes a error:  

```../../node_modules/postgres/src/connection.js:1:0
Module not found: Can't resolve 'net'

Import trace for requested module:
../../node_modules/postgres/src/index.js
../../packages/db/src/index.ts
../../packages/validators/src/index.ts
./src/app/page.tsx```

**Comment by dBianchii** [comment]
Yeah I'm getting the same problem trying to get my schemas that came from drizzle-zod in client components. I believe that drizzle-zod will only work server-side.

**Comment by juliusmarminge** [comment]
With a little different split it's possible: https://github.com/t3-oss/create-t3-turbo/pull/989

**Reply by trevorpfiz** [reply]
looks good, thanks! am I right in thinking that the drizzle-zod schemas can still not be used in the Expo app?

**Reply by juliusmarminge** [reply]
I think you should be safe to add the db as a production dependency in your expo app and use drizzle-zod that way. 

Might wanna be careful if you have a bunch of "sensitive" business logic in the db package though since i dont think Metro does treeshaking yet.

You should probably try and experiment with it yourself to be certain though
